rama-tcp/src/client/connect.rs

```diff
@@ -274,39 +275,31 @@ async fn tcp_connect_inner_branch<Dns, Connector>(
     Dns: DnsResolver<Error: Into<BoxError>> + Clone,
     Connector: TcpStreamConnector<Error: Into<BoxError> + Send + 'static> + Clone,
 {
-    /*TODO
-    modify the match to use dns.ip_mode and match for 
-    SingleIpv4 or DualPreferIpv4 or Dual
-     */
+    
     let ip_it = match ip_kind {
-        IpKind::Ipv4 => match dns.ipv4_lookup(domain).await {
+        IpKind::Ipv4 if matches!(dns.mode, DnsResolveIpMode::Dual |DnsResolveIpMode::SingleIpV4 |DnsResolveIpMode::DualPreferIpV4 ) =>{
```

Comment:
I would suggest to do:

```
match ip_kind => {
     IpKind::Ipv4 => if dns.mode.ipv4_supported() {
           // ... logic
     } else {
          trace::debug ipv4 not supported by DnsMode: no connection can be established
     }

     IpKind::Ipv6 => if dns.mode.ipv6_supported() {
           // ... logic
     } else {
          trace::debug ipv6 not supported by DnsMode: no connection can be established
     }
```

This way we have a clear trace in our logs / spans.

Btw, I don't think this part of the code is to use the Dns mode as no dns is to be resolved here. Here you need to use the IpMode or however we decided to call it.